### DISCUSSION TITLE
I'm missing something with npm run dev

### DISCUSSION THREAD

**Discussion by rernest7** [pvtl/docker-dev#33] [Q&A]
Hello and sorry for my potential ignorance, but I can't crack this one.
I can't understand how to access npm server from browser.
Dev server starts, but neither the http://localhost:<port>, or either <app folder>.localhost:<port> works. 
Adding --host flag to npm run dev changes nothing
App is laravel, inertiajs, vuejs, if that changes anything.

**Comment by wireblue** [comment]
We make sure that any Node.js tooling (which exposes ports) is run outside of our Docker LDE containers. The reason for this is that each Docker container (eg. PHP-FPM 8.3) will only expose the bare minimum ports needed to do it's job, nothing more.

This is how we run Vite on our Laravel projects.

I hope I understood your question correctly.

**Reply by rernest7** [reply]
Thanks for your response!
Are those tools simply installed on a host? I hope this question is not too invasive.

**Reply by wireblue** [reply]
Yes, we install Node.js on our host and run it natively there.

For macOS this is usually done using Homebrew. On Windows, that's done in Ubuntu linux on WSL2.
